libcloud/common/base.py

```diff
@@ -76,6 +76,39 @@
 RETRY_FAILED_HTTP_REQUESTS = False
 
 
+class LazyObject(object):
+    """An object that doesn't get initialized until accessed."""
+
+    @staticmethod
+    def _proxy(lazy_cls, *lazy_init_args, **lazy_init_kwargs):
+        class Proxy(lazy_cls, object):
+            lazy_obj = None
+
+            def __init__(self):
+                pass
+
+            def __getattribute__(self, attr):
+                lazy_obj = object.__getattribute__(self, 'lazy_obj')
+                if lazy_obj is None:
+                    lazy_obj = lazy_cls(*lazy_init_args, **lazy_init_kwargs)
+                    object.__setattr__(self, 'lazy_obj', lazy_obj)
+                return lazy_obj.__getattribute__(attr)
```

Comment:
I believe it's generally preferable to use the ```getattr``` function than invoking ```__getattribute__``` directly on a foreign object.  ```return getattr(lazy_obj, attr)```